### PR TITLE
Resolve node port conflict for internal docker registrey

### PR DESCRIPTION
**Pull request by dbadura**
Changes:
- add action which solves the node port conflict.

### CODE REVIEW

File: pkg/reconciler/instances/serverless/nodeport.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	return minRange + number
+}
+
+func setNodePortOverride(overrides map[string]interface{}, path string, port int32) {
+	overrides[path] = port
+}
+
+func getAllNodePortServices(ctx context.Context, k8sClient kubernetes.Interface, namespace string) (*corev1.ServiceList, error) {
+
+	svcs, err := k8sClient.CoreV1().Services(namespace).List(ctx, metav1.ListOptions{})
```
Comment (dbadura) [inline pkg/reconciler/instances/serverless/nodeport.go:132]: You should get all services from everynamespace

File: pkg/reconciler/instances/serverless/nodeport.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	if svc != nil {
+		if isDefaultNodePortValue(svc) {
+			return nil
+		}
+		currentNodePort := getNodePort(svc)
+		setNodePortOverride(svcCtx.Task.Configuration, dockerRegistryNodePortPath, currentNodePort)
+		return nil
+	}
+
+	svcs, err := getAllNodePortServices(svcCtx.Context, k8sClient, svcCtx.Task.Namespace)
```
Comment (kwiatekus) [inline pkg/reconciler/instances/serverless/nodeport.go:53]: This should look up node port svc from all namespaces.

**Review by kwiatekus — CHANGES_REQUESTED** [review]

File: pkg/reconciler/instances/serverless/nodeport.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("while checking if %s service is installed on cluster", dockerRegistryService))
+	}
+
+	if svc != nil {
+		if isDefaultNodePortValue(svc) {
+			return nil
+		}
+		currentNodePort := getNodePort(svc)
+		setNodePortOverride(svcCtx.Task.Configuration, dockerRegistryNodePortPath, currentNodePort)
```
Comment (kwiatekus) [inline pkg/reconciler/instances/serverless/nodeport.go:49]: shouldn't the new nodeport value be drawn in this case? I tested the code and it keeps applying the same nodeport ```
kubeClient failed to update Service 'serverless-docker-registry' (namespace: kyma-system)  with strategy 'PATCH': cannot patch "serverless-docker-registry" with kind Service: Service "serverless-docker-registry" is invalid: spec.ports[0].nodePort: Invalid value: 32137: provided port is already allocated
```

**Review by kwiatekus — CHANGES_REQUESTED** [review]

File: pkg/reconciler/instances/serverless/nodeport.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+	k8sClient, err := svcCtx.KubeClient.Clientset()
+	if err != nil {
+		return errors.Wrap(err, "while getting clientset")
+	}
+	svc, err := getService(svcCtx.Context, k8sClient, svcCtx.Task.Namespace, dockerRegistryService)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("while checking if %s service is installed on cluster", dockerRegistryService))
+	}
+
+	if svc != nil {
```
Comment (kwiatekus) [inline pkg/reconciler/instances/serverless/nodeport.go:44]:
```suggestion
	if svc != nil && isNodePortType(svc) {
```

**Review by kwiatekus — APPROVED** [review]

**Pull request merged**